build-coordinator/src/main/java/org/jboss/pnc/core/builder/coordinator/BuildCoordinator.java

```diff
@@ -57,17 +59,17 @@
 @ApplicationScoped
 public class BuildCoordinator {
 
-    private Logger log = LoggerFactory.getLogger(BuildCoordinator.class);
+    private final Logger log = LoggerFactory.getLogger(BuildCoordinator.class);
 
     /**
      * Build tasks which are either waiting to be run or currently running.
      * The task is removed from the queue when the build is complete and the results
      * are stored to the database.
      */
-    private Queue<BuildTask> activeBuildTasks = new ConcurrentLinkedQueue<>(); //TODO garbage collector (time-out, error state)
+    private final Queue<BuildTask> activeBuildTasks = new ConcurrentLinkedQueue<>(); //TODO garbage collector (time-out, error state)
```

Comment:
Looking at it I think there is no reason to use queue, it could be distributed set. Based on JDG cache?
In order to make it work properly in cluster (queue or cache) we must make sure all changes (buildTaks.status) are pushed into grid and all updates notified back.